qdrant-landing/content/documentation/tutorials/self-query.md

```diff
@@ -0,0 +1,226 @@
+
+# Load a sentence embedding model (adjust to your needs)
+encoder = SentenceTransformer('all-MiniLM-L6-v2')
+
+# Encode the wine descriptions into 512-dimension vectors
+wines['vector'] = wines['description'].apply(lambda x: encoder.encode(x).tolist())
+
+# Show the vectorized dataset
+print(wines[['description', 'vector']].head())
+```
+
+For each wine review, the `description` field is encoded into a dense vector that captures the semantic meaning of the text. The `vector` field now holds 512-dimensional vectors, which Qdrant will use for efficient similarity search.
+
+---
+
+### Step 4: Structuring the Data
+
+Each review will be stored as a document containing both the **vector** (encoded description) and **metadata** (such as country, points, and price). We’ll define a helper class to structure the data.
+
+```python
+# Helper class to structure documents
+class Document:
+    def __init__(self, page_content, metadata):
+        self.page_content = page_content
+        self.metadata = metadata
+
+# Convert DataFrame rows into Document objects
+def df_to_documents(df):
+    documents = []
+    for _, row in df.iterrows():
+        metadata = {
+            "country": row["country"],
+            "points": row["points"],
+            "price": row["price"],
+            "title": row["title"],
+            "variety": row["variety"]
+        }
+        document = Document(page_content=row["description"], metadata=metadata)
+        documents.append(document)
+    return documents
+
+# Convert wine reviews into documents
+docs = df_to_documents(wines)
+
+# Print the first document
+print(docs[0].metadata)
+```
+
+Each document consists of the encoded **description** (as `page_content`) and metadata such as **country**, **price**, and **points**. This metadata will help in filtering and fine-tuning queries later on.
+
+---
+
+### Step 5: Uploading Data to Qdrant
+
+Now that we’ve encoded the wine reviews into vectors and structured the metadata, we’ll upload the data points into the Qdrant collection. Each point will include a vector and its corresponding metadata.
+
+```python
+# Create points for each wine review
+points = [
+    models.PointStruct(
+        id=idx, 
+        vector=doc.page_content,  # Encoded vector
+        payload={'metadata': doc.metadata}  # Metadata
+    )
+    for idx, doc in enumerate(docs)
+]
+
+# Upload the points to Qdrant
+client.upload_points(
+    collection_name=collection_name,
+    points=points
+)
+```
+
+Each wine review is uploaded as a point in the collection, with both the vector representation and metadata attached to it.
+
+---
+
+### Step 6: Self-Query Retrieval in Action
+
+Let’s use **self-query retrieval** to allow the system to infer relevant filters and conditions from a natural language query. Suppose we want to find US wines priced between $15 and $30 with a rating (points) above 90. Instead of manually constructing the query, the system will automatically parse these constraints.
```

Comment:
I found this to be a problem.
The tutorial doesn't show how to infer these filters at all.